criu/pidfd.c

```diff
@@ -23,11 +25,19 @@ struct pidfd_info {
 	struct file_desc d;
 };
 
+struct dead_count {
```

Comment:
honestly, I don't understand what is going on here and I don't want to turn on the sherklock mode to figure out what you mean here. Pls write comments and a detailed commit message.

Reply:
why one process can't create a helper, open all required pidfd-s and send them to others? Cc: @mihalicyn

Reply:
Here is a draft version of what I mean in the previous comment:
https://github.com/avagin/criu/commit/f14adde2c4d7da5ad8cd6066c4867e6d6130c420

Reply:
This is a much better approach to solve this problem! Thanks @avagin 
Shouldn't the allocation for `dead` still be `shmalloc()` ? https://github.com/avagin/criu/commit/f14adde2c4d7da5ad8cd6066c4867e6d6130c420#diff-0b8ec9ac559a1c3dc6868b3f4ffaae306e633c93c387fc5867ddc83922546b9eR278

Reply:
All dead structures are allocated from one process before forking other processes, the we just read them. I don't see any reason to use shmalloc.